src/java.desktop/share/classes/javax/swing/JTabbedPane.java

```diff
@@ -1598,6 +1598,11 @@ public void setComponentAt(int index, Component component) {
             boolean selectedPage = (getSelectedIndex() == index);
 
             if (selectedPage) {
+                if (this.visComp != null &&
+                        !this.visComp.equals(component)) {
+                    // previous component visibility is set to false
```

Comment:
It seems `isVisible` is checked before changing the visibility..Should we do it here too?

Reply:
I thought it might be redundant since the setVisible() checks for visibility of component within show() and hide(). But to add clarity I have add `isVisible` check.

Reply:
It seems this check is followed in this class
https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/javax/swing/JTabbedPane.java#L390
https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/javax/swing/JTabbedPane.java#L419

Reply:
I agree. The isVisible check is used for similar purposes so it's for the best to follow convention here and use it.